daml-script/daml3/Daml/Script/Questions/Exceptions.daml

```diff
@@ -0,0 +1,46 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE InstanceSigs #-}
+
+module Daml.Script.Questions.Exceptions where
+
+#ifdef DAML_EXCEPTIONS
+import DA.Bifunctor (first)
+import DA.Exception
+import DA.Optional
+import Daml.Script.Internal
+import Daml.Script.Free (Free (..))
+
+data Catch = Catch with
+  act : () -> Free ScriptF (LedgerValue, ())
+instance IsQuestion Catch (Either AnyException x)
+
+tryToEither : forall t. (() -> Script t) -> Script (Either AnyException t)
+tryToEither act = lift Catch with act = \() -> fmap (first toLedgerValue) $ runScript (act ()) ()
```

Comment:
The act manipulation logic is upsetting, ideally I'd just pass up a `Script LedgerValue` in `Catch`, since it already contains a continuation. However `error` doesn't create a Script, it creates a direct primative call to EThrow, so the Script itself must be computed lazily.